test/common/algos.mocha.coffee

```diff
@@ -185,7 +185,44 @@ describe 'User', ->
     expect(user.stats.buffs.str).to.be 1
     expect(user.achievements.perfect).to.be 2
 
+  describe 'Resting in the Inn', ->
+    user = null
+    cron = null
+    beforeEach ->
+      user = newUser()
+      user.preferences.sleep = true
+      cron = -> user.lastCron = moment().subtract(1, 'days');user.fns.cron()
+      user.dailys = []
+      _.times 2, -> user.dailys.push shared.taskDefaults({type:'daily'})
+    it 'remains in the inn on cron', ->
+      cron()
+      expect(user.preferences.sleep).to.be.ok
+
+    it 'resets dailies', ->
+       user.dailys[0].completed = true
+       cron()
+       expect(user.dailys[0].completed).to.not.be.ok
+
+    it 'does not damage user for incomplete dailies', ->
+      expect(user).toHaveHP 50
+      user.dailys[0].completed = true
+      user.dailys[1].completed = false
+      cron()
+      expect(user).toHaveHP 50
+
+    it 'gives credit for complete dailies', ->
```

Comment:
The rest all look awesome, I just forgot to mention this before in my last round of comments. 

Similarly to checking the hp before hand, we should also check the history length before hand. Further, I'd use the expect parameter `.empty`, so the whole thing all together would look like this:

```coffeescript
it 'gives credit for complete dailies', ->
      user.dailys[0].completed = true
      expect(user.dailys[0].history).to.be.empty
      cron()
      expect(user.dailys[0].history).to.be.not.empty
```

Make that update and it'll be ready to merge in!